components/service/fretboard/README.md

```diff
@@ -12,6 +12,213 @@ Use gradle to download the library from JCenter:
+    object : RegionProvider {
+        return custom_region
+    }
+)
+```
+
+
+#### Using Kinto as experiment source
+Fretboard includes a default source implementation for a Kinto backend, which you can use like this:
+
+```Kotlin
+val fretboard = Fretboard(
+    KintoExperimentSource(baseUrl, bucketName, collectionName),
+    experimentStorage
+)
+```
+
+It expects the experiments in the following JSON format:
+```json
+{
+    "data":[
+        {
+            "name": "",
+            "match":{
+                "lang":"",
+                "appId":"",
+                "regions":[],
+                "country":"",
+                "version":"",
+                "device":"",
+                "manufacturer":"",
+                "region":""
+            },
+            "buckets": {
+                "min": "0",
+                "max": "100"
+            },
+            "description":"",
+            "id":"",
+            "last_modified":1523549895713
+        }
+    ]
+}
+```
+
+#### Using a JSON file as experiment storage
+Fretboard includes support for flat JSON files as storage mechanism out of the box:
+
+```Kotlin
+val fretboard = Fretboard(
+    experimentSource,
+    FlatFileExperimentStorage(File(context.filesDir, "experiments.json"))
+)
+```
+
+#### Creating custom experiment source
+You can create a custom experiment source simply by implementing the ExperimentSource interface:
+
+```Kotlin
+class MyExperimentSource : ExperimentSource {
+    override fun getExperiments(syncResult: SyncResult): SyncResult {
+        // ...
+        return updatedExperiments
+    }
+}
+```
+
+The getExperimentsMethod takes a SyncResult object, which contains the list of already downloaded experiments and 
+a last_modified date, and returns another SyncResult object with the updated list of experiments.
+
+As the getExperiments receives the list of experiments from storage and a last_modified date, it allows you
+to do diff requests, if your storage mechanism supports it (like Kinto does).
+
+#### Creating custom experiment storage
+You can create a custom experiment storage simply by implementing the ExperimentStorage interface, overriding
+the save and retrieve methods, which use SyncResult objects with the list of experiments and a last_modified date:
+
+```Kotlin
+class MyExperimentStorage : ExperimentStorage {
+    override fun save(syncResult: SyncResult) {
+        // save result to disk
+    }
+
+    override fun retrieve(): SyncResult {
+        // load result from disk
+        return result
+    }
+}
+```
+
+### Updating experiment list
+Fretboard provides two ways of updating the downloaded experiment list from the server: the first one is to directly
+call updateExperiments on a Fretboard instance, which forces experiments to be updated immediately and synchronously
+(do not call this on the main thread), like this:
+
+```Kotlin
+fretboard.updateExperiments()
+```
+
+The second one is to use the provided JobScheduler-based scheduler, like this:
```

Comment:
Could we make a WorkManager implementation available? WorkManager is the best solution available for scheduled jobs to date on Android. It works on any API level and even allows queueing jobs into a chain of tasks to perform.

Reply:
WorkManager still seems to be only available in an Alpha version :(

I'll file an issue though. We already have one for the telemetry library: #160

Reply:
#493